llvm/lib/IR/DIBuilder.cpp

```diff
@@ -988,9 +988,11 @@ Instruction *DIBuilder::insertDbgValueIntrinsic(Value *V,
                                                 DIExpression *Expr,
                                                 const DILocation *DL,
                                                 Instruction *InsertBefore) {
-  return insertDbgValueIntrinsic(
+  Instruction *DVI = insertDbgValueIntrinsic(
       V, VarInfo, Expr, DL, InsertBefore ? InsertBefore->getParent() : nullptr,
       InsertBefore);
+  cast<CallInst>(DVI)->setTailCall();
```

Comment:
Would it be better to put this change in `DIBuilder::insertDbgIntrinsic`, which seems to be the common leaf between all the `insertDbgXIntrinsic` functions (that's where the `IRBuilder::CreateCall` call is)?

Reply:
Probably in the long term (if dbg.values had a long term), however I think that's common between all DbgVariableIntrinsics, thus including dbg.declare too, which I don't want to get into at this stage.

Reply:
It seems harmless to cover dbg.declares too to me, given

> This change has no meaningful effect on the compiler

But I won't insist